### PR TITLE
Ae loom box2 d contact support

### PR DESCRIPTION
**Pull request by LukeLamothe**

### CODE REVIEW

File: loom/vendor/box2d/Dynamics/b2Body.cpp [hunk 1]
```diff
@@ -520,6 +520,388 @@ void b2Body::SetFixedRotation(bool flag)
...
+
+	return false; 
+}
+
+float32 b2Body::GetContactRestitution(int contactIndex)
+{
+	b2ContactEdge* edge = m_contactList;
+	int currentContactIndex = 0;
+
+	while (edge)
```
Comment (bengarney) [inline loom/vendor/box2d/Dynamics/b2Body.cpp]: Where does this increment the currentContactIndex?

**Comment by bengarney** [comment]
Rebuilding, should be done in 15 or so.

**Comment by bengarney** [comment]
And built. Gonna try it.

**Comment by bengarney** [comment]
OK, it works nicely, and the demo is awesome... BUT!

It's reporting isTouching when I'm in the AABB of shapes, not when I have an actual contact. I checked IsContacting() and it doesn't use the flag specific on the contact to detect if contact is occurring: https://github.com/LoomSDK/LoomSDK/blob/master/loom/vendor/box2d/Dynamics/Contacts/b2Contact.h#L90

So I think it should be called isTouching and check that flag.

The symptoms I see are that sloped blocks turn red when I am near them, not touching them, and I can double or triple jump if I am in the "near" zone that triggers the red.

**Comment by bengarney** [comment]
Other than that this looks green to land.

**Comment by LukeLamothe** [comment]
Nice catch!  OK, pretty sure I've fixed that.  I've added a param for the
relevant flags, defaulting them to Enabled|Touching.

Pushed and ready :)

Luke

**Comment by bengarney** [comment]
Code looks good, testing it on my Mac.

**Comment by bengarney** [comment]
Runs well. A little glitchy but it appears to only be due to Box2D finickiness/naive jumping logic not collision detection.

**Pull request merged**